### DISCUSSION TITLE
Poweroff hook

### DISCUSSION THREAD

**Discussion by seebeen** [ddev/ddev#5269] [Ideas]
Is there a poweroff hook?
Or does `post-stop` cover that?

I have an addon I'm working on, which injects another global container, and I would like to stop and destroy it on poweroff.

**Comment by rfay** [comment]
That's an interesting application! There are no poweroff hooks, although they'd be easy to add and a PR would be welcome. But you could easily add a post-stop hook that queried whether the ddev-ssh-agent was still running, for example, and if not, also stop your container.

**Comment by seebeen** [comment]
I'm finalizing something that I believe you'll find interesting.
I'll keep you posted, so we can discuss further 😊

**Reply by seebeen** [reply]
Or, better yet - here is a Work in Progress.

It's a global container serving as a redis gui for all of redis instances in the cluster.
https://github.com/oblakstudio/ddev-redis-insight/tree/develop

**Reply by rfay** [reply]
Cool. Mostly people prefer individual instances for everything. Of course this is a UI, so that may be a little different. 

I don't understand why you're adding traefik configuration with a hook, instead of just installing it directly.

**Reply by seebeen** [reply]
How can I install it directly - didn't find it in the docs? 😊

**Reply by rfay** [reply]
You can install any file, just the same as you're currently doing with the files you're installing. https://github.com/oblakstudio/ddev-redis-insight/blob/develop/install.yaml has global_files and project_files. Just add your traefik config in there.

**Reply by seebeen** [reply]
OK, thanks!

**Reply by seebeen** [reply]
Wherever I add the traefik config, it either magically dissapears, or it's not loaded / read :(

I need to add it only once, because I'm defining a custom subdomain.

**Reply by rfay** [reply]
You would add it on installation of the add-on. But unless you remove the `#ddev-generated` it will, of course, be removed.

**Reply by seebeen** [reply]
And that would prevent upgrades, Or I would have to use sed to modify the file.

Would a PR to read custom files from the global trafik dir be welcome?
I assume it's golang?

**Reply by rfay** [reply]
DDEV already supports global traefik config.

**Reply by rfay** [reply]
https://ddev.readthedocs.io/en/latest/users/extend/traefik-router/#global-traefik-configuration
